packages/react-components/src/components/DevicePermissions/DomainPermissions.tsx

```diff
@@ -23,6 +23,10 @@ export interface CommonDomainPermissionsProps {
    * Type of the Domain Permissions component.
    */
   type: 'request' | 'denied' | 'check';
+  /**
+   * Whether we are using safari browser or not
+   */
+  isSafari?: boolean;
```

Comment:
`isSafari` is OK, but if we ever needed to do something specific for firefox having a boolean makes it less extensible

Reply:
If we want to move the logic to callreadinessmodal and pass it down to domain permissions, we can only pass along a boolean. Earlier I had the logic inside domain permissions component so that when we need to do something for firefox it also works

Reply:
Why is it we can only pass a boolean? We can do
```
const isSafari = _isSafari(environmentInfo);
<CameraAndMicrophoneDomainPermissions
  browser={isSafari ? 'safari' | 'other'}
/>
```

Reply:
In this way we are still limited when we need to do something specific for firefox, since 'other' includes chrome and firefox. Should we have the browser type to be safari | firefox | chrome | edge ?

Reply:
When we need to do firefox we can then add it, but you're correct that `other` isn't so great, perhaps:
`browserHint: 'safari' | 'unset'`
?

Reply:
So I like the idea of spreading it out to define the major browsers explicitly so `other` refers to browsers like Brave, duckduckgo and so on. I'd argue Opera might be in there too but it is prominent enough in the world that Calling could want to support it in the distant future.